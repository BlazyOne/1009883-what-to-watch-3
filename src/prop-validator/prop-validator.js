import PropTypes from 'prop-types';

const PropValidator = {
  PROMO_FILM: PropTypes.exact({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired
  }),
  FILMS: PropTypes.arrayOf(
      PropTypes.exact({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired
      })
  ),
  FILM: PropTypes.exact({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }),
  FILM_INFO: PropTypes.exact({
    id: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    ratingScore: PropTypes.string.isRequired,
    ratingLevel: PropTypes.string.isRequired,
    ratingCount: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.string.isRequired
  }),
  ON_TITLE_CLICK: PropTypes.func.isRequired,
  ON_CARD_CLICK: PropTypes.func.isRequired,
  ON_MOUSE_OVER_CARD: PropTypes.func.isRequired
};

export {PropValidator};
