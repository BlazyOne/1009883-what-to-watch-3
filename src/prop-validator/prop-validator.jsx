import PropTypes from 'prop-types';

const PropValidator = {
  PROMO_FILM: PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired
  }),
  FILMS: PropTypes.arrayOf(PropTypes.string.isRequired),
  ON_TITLE_CLICK: PropTypes.func.isRequired
};

export {PropValidator};
