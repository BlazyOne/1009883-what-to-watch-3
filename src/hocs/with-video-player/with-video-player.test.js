import React from 'react';
import renderer from 'react-test-renderer';
import withVideoPlayer from './with-video-player.jsx';
import {PropValidator} from '../../prop-validator/prop-validator.js';

const mockVideoSettings = {
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  width: `280`,
  height: `175`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  muted: true,
  looped: true,
  stopOnPause: true,
  videoClass: ``
};

const MockComponent = (props) => {
  const {renderVideoPlayer} = props;

  return (
    <React.Fragment>
      {renderVideoPlayer(mockVideoSettings)}
    </React.Fragment>
  );
};

MockComponent.propTypes = {
  renderVideoPlayer: PropValidator.RENDER_VIDEO_PLAYER
};

const MockComponentWrapped = withVideoPlayer(MockComponent);

it(`Should withVideoPlayer render correctly`, () => {
  const tree = renderer.create(
      <MockComponentWrapped/>, {
        createNodeMock() {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
