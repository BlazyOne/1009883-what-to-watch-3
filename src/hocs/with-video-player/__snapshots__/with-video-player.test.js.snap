// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withVideoPlayer render correctly 1`] = `
<video
  className=""
/>
`;
